sdk/nodejs/cmd/pulumi-language-nodejs/language_test.go

```diff
@@ -33,6 +36,53 @@ import (
 	"google.golang.org/grpc/credentials/insecure"
 )
 
+type hostEngine struct {
+	pulumirpc.UnimplementedEngineServer
+	t *testing.T
+}
+
+func (e *hostEngine) Log(_ context.Context, req *pulumirpc.LogRequest) (*pbempty.Empty, error) {
+	var sev diag.Severity
+	switch req.Severity {
+	case pulumirpc.LogSeverity_DEBUG:
+		sev = diag.Debug
+	case pulumirpc.LogSeverity_INFO:
+		sev = diag.Info
+	case pulumirpc.LogSeverity_WARNING:
+		sev = diag.Warning
+	case pulumirpc.LogSeverity_ERROR:
+		sev = diag.Error
+	default:
+		return nil, fmt.Errorf("Unrecognized logging severity: %v", req.Severity)
+	}
+
+	if req.StreamId != 0 {
+		e.t.Logf("(%d) %s[%s]: %s", req.StreamId, sev, req.Urn, req.Message)
+	} else {
+		e.t.Logf("%s[%s]: %s", sev, req.Urn, req.Message)
+	}
+	return &pbempty.Empty{}, nil
+}
+
+func runEngine(t *testing.T) string {
```

Comment:
I'm not 100% convinced this is the way we want to do this engine logging but it'll work for now. Something to keep thinking about.